src/test/random/CMakeLists.txt

```diff
@@ -1,8 +1,6 @@
-## build the test as a dynamic executable that plugs into shadow
-add_executable(shadow-plugin-test-random test_random.c ../test_common.c)
-
 ## register the tests
-add_test(NAME random COMMAND shadow-plugin-test-random)
+add_test(NAME random COMMAND ../../target/debug/test_random --libc-passing)
+list(APPEND TestNames random)
 
 add_test(
     NAME random-shadow-ptrace
```

Comment:
You should add the shadow tests (random-shadow-ptrace and random-shadow-preload) to the `TestNames` list as well.

Also, you'll need to update the binary path in `src/test/random/random.test.shadow.config.yaml` since they will still be pointing to the old C binary path.

Reply:
Not sure if the path in `random.test.shadow.config.yaml` is correct, yet. The `CMakeLists.txt` now points  to the correct path.